lib/src/ruby/module.cc

```diff
@@ -953,16 +963,16 @@ namespace facter { namespace ruby {
 
         // Look for "facter" subdirectories on the load path
         for (auto const& directory : ruby.get_load_path()) {
-            // Get the absolute directory name
             boost::system::error_code ec;
-            path dir = absolute(directory);
+            auto dir = canonicalize(directory).string();
```

Comment:
Since this is now a string, you need to remove the `dir.string` below on line 979

Reply:
Made a few more changes.